src/context/WalletProvider/KeepKey/hooks/useKeepKeyEventHandler.ts

```diff
@@ -242,11 +246,14 @@ export const useKeepKeyEventHandler = (
       }
     }
 
-    // Handle all KeepKey events
-    keyring.on(['KeepKey', '*', '*'], handleEvent)
-    // HDWallet emits (DIS)CONNECT events as "KeepKey - {LABEL}" so we can't just listen for "KeepKey"
-    keyring.on(['*', '*', Events.CONNECT], handleConnect)
-    keyring.on(['*', '*', Events.DISCONNECT], handleDisconnect)
+    // We only want to listen to these events if a KeepKey is actually connected, or we are showing a KeepKey modal
+    if ([state.connectedType, state.modalType].includes(KeyManager.KeepKey)) {
+      // Handle all KeepKey events
+      keyring.on(['KeepKey', '*', '*'], handleEvent)
+      // HDWallet emits (DIS)CONNECT events as "KeepKey - {LABEL}" so we can't just listen for "KeepKey"
+      keyring.on(['*', '*', Events.CONNECT], handleConnect)
```

Comment:
This was a big rug - it was firing whenever MetaMask was connected, and setting the connected wallet to KeepKey.